components/bl-emoji/src/index.js

```diff
@@ -0,0 +1,71 @@
+import { useState, useMemo } from 'react';
+
+import EmojiPicker from './emoji-picker-react.min.js';
+
+const { cn, normalizeDimensionValue } = BackendlessUI.CSSUtils;
+
+export default function EmojiComponent({ component, eventHandlers, elRef }) {
+  const {
+    classList, style, display, disabled, buttonColor, buttonSize,
+    dropdownHeight, dropdownWidth, dropdownPosition, theme, emojiStyle, searchDisabled
+  } = component;
+  const { onEmojiSelect } = eventHandlers;
+
+  const [pickerVisibility, setPickerVisibility] = useState(false);
+
+  const validButtonSize = useMemo(() => normalizeDimensionValue(buttonSize), [buttonSize]);
+  const validDropdownWidth = useMemo(() => normalizeDimensionValue(dropdownWidth), [dropdownWidth]);
+  const validDropdownHeight = useMemo(() => normalizeDimensionValue(dropdownHeight), [dropdownHeight]);
+
+  const handleEmojiButtonClick = () => {
+    setPickerVisibility(prevState => !prevState);
+  };
+
+  const handleEmojiClick = event => {
+    const { emoji, names } = event;
+
+    navigator.clipboard.writeText(emoji);
+    onEmojiSelect({ emoji, emojiNames: names });
+  };
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div
+      ref={ elRef }
+      style={ style }
+      className={
+        cn("bl-customComponent-emoji", classList, { "bl-customComponent-emoji--disabled": disabled })
+      }>
+      <div className="emoji-picker">
+        <div onClick={ handleEmojiButtonClick } className="emoji-picker__button">
+          <svg width={ validButtonSize } height={ validButtonSize } viewBox="0 0 512 512" fill={ buttonColor }>
+            <path d={ getIconPath(pickerVisibility) } />
```

Comment:
Two icon components and something like that

{ pickerVisibility ? <IconShow> : <IconHide> }